File: problems/0004_median-of-two-sorted-arrays/concat-sort/Solution.scala
object Solution {
    def findMedianSortedArrays(nums1: Array[Int], nums2: Array[Int]): Double = {
        val nums = (nums1 ++ nums2).sorted
        val i = nums.size / 2
        if (nums.size % 2 == 0)
            (nums(i - 1) + nums(i)) / 2.0
        else
            nums(i)
    }
}

File: problems/0004_median-of-two-sorted-arrays/concat-sort/Solution.kt
class Solution {
    fun findMedianSortedArrays(nums1: IntArray, nums2: IntArray): Double {
        val nums = (nums1 + nums2).sortedArray()
        val i = nums.size / 2
        return if (nums.size % 2 == 0) (nums[i - 1] + nums[i]) / 2.0 else nums[i].toDouble()
    }
}
